### PR TITLE
GA 🎉

### PR DESCRIPTION
**Pull request by manekinekko**

### CODE REVIEW

File: src/cli/commands/start.ts [hunk 1]
```diff
@@ -102,7 +106,7 @@ Serve static content from a folder and run an API from another folder
 swa start ./output-folder --api-location ./api
 
 Use a custom command to run framework development server at startup
-swa start http://localhost:3000 --run "npm start"
+swa start http://localhost:3000 --run-build "npm start"
```
Comment (manekinekko) [inline src/cli/commands/start.ts:109]: this should be `--run` not `--run-build`

File: src/config.ts [hunk 2]
```diff
@@ -34,10 +33,10 @@ export const DEFAULT_CONFIG: SWACLIConfig = {
...
   host: SWA_CLI_HOST || "localhost",
   apiPort: parseInt(SWA_CLI_API_PORT || "7071", 10),
-  appLocation: SWA_CLI_APP_LOCATION || `.${path.sep}`,
+  appLocation: SWA_CLI_APP_LOCATION || `.`,
   apiLocation: SWA_CLI_API_LOCATION ? SWA_CLI_API_LOCATION : undefined,
-  outputLocation: SWA_CLI_OUTPUT_LOCATION || `.${path.sep}`,
-  swaConfigLocation: SWA_RUNTIME_CONFIG_LOCATION || `.${path.sep}`,
+  outputLocation: SWA_CLI_OUTPUT_LOCATION || `.`,
+  swaConfigLocation: SWA_RUNTIME_CONFIG_LOCATION || `.`,
   ssl: useEnvVarOrUseDefault(SWA_CLI_APP_SSL, false),
```
Comment (manekinekko) [inline src/config.ts:41]: default paths should match those expected by SWA

File: src/config.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ const {
   SWA_CLI_APP_NAME,
   SWA_CLI_DEPLOY_DRY_RUN,
   SWA_CLI_DEPLOY_ENV,
+  SWA_CLI_LOGIN_CLEAR_CREDENTIALS
```
Comment (manekinekko) [inline src/config.ts]: we need to add this to ENV.md

File: .github/workflows/node.yml [hunk 4]
```diff
@@ -9,17 +9,19 @@ on:
   pull_request:
     branches: [main, ga]
 
+env:
+  NODE_VERSIONS: [14.x, 15.x, 16.x, 17.x]
```
Comment (manekinekko) [inline .github/workflows/node.yml]: Node 18 is now OUT. We should test against it (if possible).

File: src/core/frameworks/detect.ts [hunk 5]
```diff
@@ -319,7 +319,7 @@ function orderFrameworksByParent(detectedFolders: DetectedFolder[]): void {
 }
 
 function findAllFiles(fileglob: string, files: string[]): string[] {
-  const { regex } = globrex(`?(*/)${fileglob}`, { extended: true, flags: "i" } as globrex.Options);
+  const { regex } = globrex(`?(*${path.sep})${fileglob}`, { extended: true, flags: "i" } as globrex.Options);
```
Comment (manekinekko) [inline src/core/frameworks/detect.ts:322]: Why not use the existing `globalyzer` module instead of `globrex`?

File: src/core/frameworks/frameworks.ts [hunk 6]
```diff
@@ -41,7 +41,7 @@ export const appFrameworks: FrameworkDefinition[] = [
     id: "static",
     name: "Static HTML",
     config: {
-      outputLocation: "./",
+      outputLocation: ".",
```
Comment (manekinekko) [inline src/core/frameworks/frameworks.ts:44]: what's the reason for these changes?

**Comment by sinedied** [comment]
@manekinekko could you have a look at commit 80503fefa7bbf85b01f639be526f0c3caefdee1c ?

File: src/core/frameworks/detect.ts (hunk 5 above)
Comment (manekinekko) [reply src/core/frameworks/detect.ts:322]: TODO: remove globalyzer.

File: src/cli/commands/start.ts [hunk 7]
```diff
@@ -1,77 +1,208 @@
...
+    .option("--host <host>", "the host address to use for the CLI dev server", DEFAULT_CONFIG.host)
+    .option<number>("--port <port>", "the port value to use for the CLI dev server", parsePort, DEFAULT_CONFIG.port)
+    .option("--run-build", 'run "swa build" before starting the emulator', false)
+
+    .option("--ssl", "serve the front-end application and API over HTTPS", DEFAULT_CONFIG.ssl)
+    .option("--ssl-cert <sslCertLocation>", "the SSL certificate (.crt) to use when enabling HTTPS", DEFAULT_CONFIG.sslCert)
+    .option("--ssl-key <sslKeyLocation>", "the SSL key (.key) to use when enabling HTTPS", DEFAULT_CONFIG.sslKey)
+    .option("--run <startupScript>", "run a custom shell command or script file at startup", DEFAULT_CONFIG.run)
+    .option<number>(
+      "--server-timeout <time>",
```
Comment (alexweininger) [inline src/cli/commands/start.ts:51]: Is there a reason this option was renamed from `--devserver-timeout` to `--server-timeout`? It'd be nice to keep it `--devserver-timeout` for consistency and reducing breaking changes.

Comment (alexweininger) [reply src/cli/commands/start.ts:51]: @manekinekko @sinedied

Comment (manekinekko) [reply src/cli/commands/start.ts:51]: We were discussing these changes with @sinedied during our review the other day. I think we agreed to rename this flag back to `--devserver-timeout`. We haven't had the time to revert the change.

We are also introducing these new flags:
- `--app-devserver-url`
- `--api-devserver-url`

FYI, another breaching change introduced is that the `--server-timeout` flag now accepts values in seconds, **not milliseconds**.

Comment (manekinekko) [reply src/cli/commands/start.ts:51]: @sinedied  can you revert the change in your pending PR (https://github.com/Azure/static-web-apps-cli/pull/472)?

`--server-timeout` --> `--devserver-timeout`

Comment (alexweininger) [reply src/cli/commands/start.ts:51]: > We are also introducing these new flags:
> 
> * --app-devserver-url
> * --api-devserver-url

Will these flags be replacing any existing flags, or are they totally new?

File: src/core/frameworks/detect.ts (hunk 5 above)
Comment (sinedied) [reply src/core/frameworks/detect.ts:322]: done

File: src/cli/commands/start.ts (hunk 1 above)
Comment (sinedied) [reply src/cli/commands/start.ts:107]: Fixed, `--run-build` was removed in favor or `swa build`

**Comment by manekinekko** [comment]
TODO: when calling `swa deploy --print-token`:

✖ The project "angular-auth0" is linked to "GitHub"!
✖ Unlink the project from the "GitHub" provider and try again.
